### PR TITLE
workflows: use Homebrew/actions/download-artifact

### PR DESCRIPTION
**Pull request by jonchang**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/linuxbrew-core/blob/master/CONTRIBUTING.md)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/linuxbrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?
- [ ] Have you included the output of `brew gist-logs <formula>` of the build failure if your PR fixes a build failure. Please quote the exact error message.

-----

### CODE REVIEW

File: .github/workflows/upload-bottles.yml [hunk 1]
```diff
@@ -14,96 +14,14 @@ jobs:
...
-          mkdir ~/bottles
-          cd ~/bottles
-          artifact_id=${{steps.artifact.outputs.result}}
-          echo "artifact_id=$artifact_id"
-          curl -L -o bottles.zip "https://${{secrets.HOMEBREW_GITHUB_API_TOKEN}}@api.github.com/repos/${{github.repository}}/actions/artifacts/$artifact_id/zip"
-          file bottles.zip
-          unzip bottles.zip
+          github_token: ${{secrets.HOMEBREW_GITHUB_API_TOKEN}}
+          workflow: build-bottles.yml
+          commit: ${{github.sha}}
```
Comment (dawidd6) [inline .github/workflows/upload-bottles.yml]: That won't work. The commit SHA pushed to master differs from the one in the PR. Would need to use another Action, to get the PR associated with pushed commit, like: https://github.com/jwalton/gh-find-current-pr (what we are doing currently in `github.repos.listPullRequestsAssociatedWithCommit`).

Comment (jonchang) [reply .github/workflows/upload-bottles.yml]: Good catch, thanks! I got my wires mixed up working on Homebrew/core and forgot that Linux would need a different workflow.

File: .github/workflows/upload-bottles.yml [hunk 2]
```diff
@@ -14,96 +14,28 @@ jobs:
...
-          curl -L -o bottles.zip "https://${{secrets.HOMEBREW_GITHUB_API_TOKEN}}@api.github.com/repos/${{github.repository}}/actions/artifacts/$artifact_id/zip"
-          file bottles.zip
-          unzip bottles.zip
+            return prs.data[0]
+      - name: Download workflow artifacts
+        uses: Homebrew/actions/download-artifact@master
+        with:
+          github_token: ${{secrets.HOMEBREW_GITHUB_API_TOKEN}}
+          workflow: build-bottles.yml
+          commit: ${{steps.pr.outputs.number}}
```
Comment (dawidd6) [inline .github/workflows/upload-bottles.yml]: It needs to be the `pr` input.

Comment (dawidd6) [reply .github/workflows/upload-bottles.yml]: Thinking about it now, maybe we could insert here the PR head commit sha?

Would `${{steps.pr.outputs.head.sha}}` work?

Comment (jonchang) [reply .github/workflows/upload-bottles.yml]: Oops, good catch. Would you prefer the `pr` input or the `commit` input? I feel like it should function identically either way?

Comment (dawidd6) [reply .github/workflows/upload-bottles.yml]: If we specify the `commit` input, we have one API call less to make, so I would say `commit` is better in that case.

Comment (jonchang) [reply .github/workflows/upload-bottles.yml]: Setting output like in https://github.com/Homebrew/homebrew-core/blob/f3a8769e6ae498ae98a46abf8b5d01647c133efb/.github/workflows/publish-commit-bottles.yml#L25-L27

Returning the JSON object directly doesn't work as it pops it out as a string that we need to run through e.g., `jq`

File: .github/workflows/upload-bottles.yml [hunk 3]
```diff
@@ -14,101 +14,34 @@ jobs:
     container:
       image: homebrew/brew
     steps:
-      - name: Get artifact data
+      - name: Determine associated pull request
         uses: actions/github-script@0.8.0
```
Comment (dawidd6) [inline .github/workflows/upload-bottles.yml]:
```suggestion
        uses: actions/github-script@0.9.0
```

File: .github/workflows/upload-bottles.yml [hunk 4]
```diff
@@ -14,101 +14,34 @@ jobs:
...
-          set -eu
-          mkdir ~/bottles
-          cd ~/bottles
-          artifact_id=${{steps.artifact.outputs.result}}
-          echo "artifact_id=$artifact_id"
-          curl -L -o bottles.zip "https://${{secrets.HOMEBREW_GITHUB_API_TOKEN}}@api.github.com/repos/${{github.repository}}/actions/artifacts/$artifact_id/zip"
-          file bottles.zip
-          unzip bottles.zip
+            process.stdout.write("::set-output name=number::" + pr.data[0].number + "\n")
+            process.stdout.write("::set-output name=head_sha::" + pr.data[0].head.sha + "\n")
```
Comment (dawidd6) [inline .github/workflows/upload-bottles.yml]:
```suggestion
            core.setOutput("head_sha", pr.data[0].head.sha)
```

**Comment by jonchang** [comment]
Ah, they finally updated github-script! Thanks for the reviews @dawidd6!

**Pull request merged**